hpx/util/detail/pack_traversal_async_impl.hpp

```diff
@@ -596,26 +598,21 @@ namespace util {
             typename types::visitor_pointer_type
         {
             // Create the frame on the heap which stores the arguments
-            // to traverse asynchronous.
-            auto frame = [&] {
-                auto ptr = new
+            // to traverse asynchronous. It is put into an intrusive_ptr which
+            // is needed to keep the frame alive.
+            auto frame = typename types::frame_pointer_type(new
                     typename types::frame_type(std::forward<Visitor>(visitor),
-                        std::forward<Args>(args)...);
-
-                // Create an intrusive_ptr from the heap object, don't increase
-                // reference count (it's already 'one').
-                return typename types::frame_pointer_type(ptr, false);
-            }();
+                        std::forward<Args>(args)...));
```

Comment:
Wouldn't this initialize the intrusive_ptr with a ref-count of 2?

Reply:
You are correct. The interface isn't really intuitive, the future_data based frames all have in initial refcount, the one in the unit test doesn't...

Reply:
Ok, only the unit test itself is broken :/
I'm having real difficulties to fix it properly...